auth/src/main/java/io/camunda/zeebe/auth/impl/JwtAuthorizationDecoder.java

```diff
@@ -122,4 +111,25 @@ public JwtAuthorizationDecoder withJwtToken(final String token) {
     jwtToken = token;
     return this;
   }
+
+  private DecodedJWT validateJwtToken() {
+    final Verification verificationBuilder =
+        JWT.require(signingAlgorithm)
+            .withIssuer(issuer)
+            .withAudience(audience)
+            .withSubject(subject)
+            .ignoreIssuedAt();
+
+    for (final String claim : claims) {
+      verificationBuilder.withClaimPresence(claim);
+    }
+    final JWTVerifier jwtVerification = verificationBuilder.build();
+
+    try {
+      return jwtVerification.verify(jwtToken);
+    } catch (final JWTVerificationException | NullPointerException ex) {
+      LOGGER.error("Authorization data unavailable: {}", ex.getMessage());
+      throw new UnrecoverableException("Authorization data unavailable: " + ex.getMessage(), ex);
+    }
```

Comment:
🔧 The try-catch is now duplicated between here and lines 76-81. I suggest inlining this method again, as all the builder does is validation (pretty much).